### PR TITLE
Build dynamic libs instead of static ones

### PR DESCRIPTION
**Pull request by ilya-lavrenov**
Switched to dynamic libraries, since package managers have a goal to avoid duplication of binary sizes because of state linkage.

Checklist:
* [X] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [X] Bumped the build number (if the version is unchanged)
* [X] Reset the build number to `0` (if the version changed)
* [X] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [X] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
@conda-forge-admin, please rerender

**Comment by ilya-lavrenov** [comment]
@xhochy, @SylvainCorlay and @scopatz 
Could you please review?

File: recipe/meta.yaml [hunk 1]
```diff
@@ -27,8 +27,10 @@ test:
   commands:
     - test -f "${PREFIX}/lib/libpugixml.a"  # [unix]
```
Comment (xhochy) [inline recipe/meta.yaml]:
```suggestion
    - test ! -f "${PREFIX}/lib/libpugixml.a"  # [unix]
```

File: recipe/build.sh [hunk 2]
```diff
@@ -7,10 +7,9 @@ cd build
 
 cmake ${CMAKE_ARGS} .. \
  -DCMAKE_BUILD_TYPE=Release \
- -DCMAKE_INSTALL_PREFIX=$PREFIX \
- -DCMAKE_POSITION_INDEPENDENT_CODE:BOOL=TRUE \
- -DCMAKE_INSTALL_LIBDIR=lib
+ -DBUILD_SHARED_LIBS=ON \
+ -DBUILD_SHARED_AND_STATIC_LIBS=ON || exit 1
```
Comment (xhochy) [inline recipe/build.sh]:
```suggestion
 -DBUILD_SHARED_LIBS=ON
```

File: recipe/build.sh [hunk 3]
```diff
@@ -7,10 +7,9 @@ cd build
...
 
-make -j $CPU_COUNT
+make -j $CPU_COUNT || exit 1
```
Comment (xhochy) [inline recipe/build.sh]: No need for `exit 1`; with `set -e` this will stop anyways.
```suggestion
make -j $CPU_COUNT
```

**Pull request merged**